api/search.go

```diff
@@ -35,12 +34,15 @@ func (h *SearchHandlers) searchHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	res, err := h.SearchMethods.Search(r.Context(), params)
+	got, _, err := h.Instance.Dispatch(r.Context(), "search.search", params)
 	if err != nil {
-		log.Infof("search error: %s", err.Error())
-		util.WriteErrResponse(w, http.StatusBadRequest, err)
+		util.RespondWithError(w, err)
+		return
+	}
+	res, ok := got.([]lib.SearchResult)
+	if !ok {
+		util.RespondWithDispatchTypeError(w, got)
 		return
```

Comment:
This handler can be replaced w/ `NewHTTPRequestHandler` in `api/api.go` & then we can remove this file! (Any handlers that don't adjust the return from Dispatch (and can be expressed w/ "Content-Type application/json", so no "text/csv") can be replaced by `NewHTTPRequestHandler`